db/migrate/20250204190405_create_oidc_trusted_publisher_buildkite.rb

```diff
@@ -0,0 +1,16 @@
+class CreateOIDCTrustedPublisherBuildkite < ActiveRecord::Migration[8.0]
+  disable_ddl_transaction!
+
+  def change
+    create_table :oidc_trusted_publisher_buildkites do |t|
+      t.string :organization_slug, null: false
```

Comment:
are these two slugs both immutable, or is it possible for someone to take over an organization name that was previously owned by someone else?

Reply:
* Pipeline slug is not very immutable. Similar to a github repository in an account, they can be deleted, renamed and re-use slugs
* Organization slug is unique for as long as the account exists, including if the customer cancels their account. However, if they opt into *deleting* their account then the slug does become available for others

Reply:
in that case, can we do what we do for GitHub and also store an organization_id that gets fetched on trusted publisher creation to prevent cross-organization name takeovers?

Reply:
Unfortunately we don't have a public API that would make this as easy as it is for the GitHub provider.

We do have a unique and immutable `organization_id` for each organization and it's available as a claim in our OIDC tokens. The value is a UUID. It's less readable to the user, but I can add it as a requirement here to avoid the risk of slug reuse attacks.

If we do that to prevent cross-organization risk, are you comfortable leaving `pipeline_slug` as a readable scope within an organization context?

Reply:
I _think_ that's the right tradeoff on both counts, but also curious what @woodruffw & @sethmlarson as my collaborators on the trusted publishing guides from the openssf